lib/rubycritic/analysers/coverage.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'rubycritic/colorize'
+require 'json'
+require 'simplecov'
+
+module RubyCritic
+  module Analyser
+    class Coverage
+      include Colorize
+
+      attr_reader :resultset_locked
+
+      def initialize(analysed_modules)
+        @analysed_modules = analysed_modules
+        @result = results.first
+      end
+
+      def run
+        @analysed_modules.each do |analysed_module|
+          analysed_module.coverage = find_coverage_percentage(analysed_module)
+          print green '.'
+        end
+        puts ''
+      end
+
+      def to_s
+        'simple_cov'
+      end
+
+      private
+
+      def find_coverage_percentage(analysed_module)
+        return 0 unless (source_file = find_source_file(analysed_module))
+
+        source_file.covered_percent
+      end
+
+      def find_source_file(analysed_module)
+        return unless @result
+
+        needle = File.join(SimpleCov.root, analysed_module.path)
+
+        @result.source_files.detect { |file| file.filename == needle }
+      end
+
+      # The path to the .resultset.json cache file
+      def resultset_path
+        File.join(SimpleCov.coverage_path, '.resultset.json')
+      end
+
+      def resultset_writelock
+        File.join(SimpleCov.coverage_path, '.resultset.json.lock')
+      end
+
+      # Loads the cached resultset from JSON and returns it as a Hash,
+      # caching it for subsequent accesses.
+      def resultset
+        @resultset ||= begin
+          if (data = stored_data)
+            begin
+              JSON.parse(data) || {}
+            rescue
```

Comment:
This should rescue from `JSON::ParserError` or at least `StandardError` and probably print to STDOUT.
It should not happen, but if it does the user would like to know.